File: 백준/Bronze/2903. 중앙 이동 알고리즘/중앙 이동 알고리즘.java
import java.util.Scanner;

public class Main {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int n = sc.nextInt();

        int dot[] = new int[n + 1];
        dot[0] = 4;

        for (int i = 1; i <= n; i++) {
            dot[i] = (int) Math.pow((2 * (Math.sqrt(dot[i - 1])) - 1), 2);
        }
        System.out.println(dot[n]);
    }
}

File: 백준/Bronze/2903. 중앙 이동 알고리즘/중앙 이동 알고리즘.py
import math

n = int(input())

arr = [0] * (n + 1)

arr[0] = 4

for i in range(1, n + 1):
	arr[i] = ((math.sqrt(arr[i - 1]) * 2) - 1) ** 2

print(int(arr[n]))